html5lib/treebuilders/etree.py

```diff
@@ -129,8 +130,10 @@ def insertText(self, data, insertBefore=None):
 
         def cloneNode(self):
             element = type(self)(self.name, self.namespace)
-            for name, value in self.attributes.items():
-                element.attributes[name] = value
+            try:
+                element._element.attrib = self._element.attrib.copy()
+            except AttributeError:
+                element.attributes = self.attributes
```

Comment:
It's unclear why this can throw.

Reply:
I've now split out the lxml impl of cloneNode (tl;dr: in lxml `attrib` is constant).